packages/aws-cdk/.gitignore

```diff
@@ -17,3 +17,7 @@ nyc.config.js
 
 !test/integ/run-wrappers/dist
 !test/integ/cli/**/*
+
+# Might be created when running regression tests.
+# But normally should be deleted after execution.
+test/integ/cli-backwards-tests-*
```

Comment:
Wouldn't it make more sense to create in `/tmp`?

Reply:
### Long answer to a short question

The problem is that the tests in that directory use files outside of it (for example `common.bash`, `run-against-repo`). And If I were to use the previous version of `run-against-repo` for example I wouldn't have been able to run it against published libraries. (since it doesn't support it).

The quickest way to get this running was by adding the necessary support to the wrappers and maintaining the same directory structure by having it be a sibling of the regular `integ/cli` directory.

Writing this now makes me think that running common code from one version with tests from another might be problematic. Though one can also argue that we just shouldn't make changes to that common code that break previous integration tests (this will be validated by the test naturally).

We can also incrementally change this, now that the wrappers and the scripts support what we need, we can change this in the next release. 

What do you think?

Reply:
It definitely doesn’t make sense to run the common code from the wrong version but I understand that pragmatically this makes sense now. Let’s open an issue and take care in 2025.